index.js

```diff
@@ -13,133 +13,167 @@
  * https://astexplorer.net/#/gist/7bc4771564a12c9f93c4904b3934aa1c/latests
  */
 
-module.exports = function (babel) {
-    const { types: t } = babel;
+module.exports = function(babel) {
+  const {types: t} = babel;
+
+  const fixContentProp = args => {
+    return args.map(argument => {
+      if (argument.type !== "ObjectExpression") {
+        return argument;
+      }
+      return {
+        ...argument,
+        properties: argument.properties.map(prop => {
+          if (prop.key.name !== "content") {
+            return prop;
+          }
 
-    return {
-        visitor: {
-            /**
-             * Find all JSX elements that use the old
-             * <glamorous.Something> notation and rewrite them to be
-             * plain ole somethings.
-             */
-            JSXElement(path) {
-                path.traverse({
-                    "JSXOpeningElement|JSXClosingElement"(path) {
-                        if (!t.isJSXMemberExpression(path.node.name)) {
-                            return;
-                        }
-                        if (path.node.name.object.name !== "glamorous") {
-                            return;
-                        }
-                        path.node.name = t.identifier(
-                            path.node.name.property.name.toLowerCase()
-                        );
-                    }
-                });
-            },
+          // Add quotes to the content property.
+          return {
+            ...prop,
+            value: t.stringLiteral(`"${prop.value.value}"`),
+          };
+        }),
+      };
+    });
+  };
 
-            /**
-             * Find glamorous import statements, including ThemeProvider
-             * imports and rewrite them to be emotion imports.
-             */
-            ImportDeclaration(path, { opts }) {
-                if (path.node.source.value !== "glamorous") {
-                    return;
-                }
+  const transformJSXAttributes = attrs => {
+    if (!attrs) return [];
+    const appendToCss = [];
+    let cssAttr = null;
+    const newAttrs = attrs.filter(attr => {
+      const {value, name, type} = attr;
+      if (type === "JSXSpreadAttribute") return true;
+      if (name.name === "css") {
+        cssAttr = attr;
+      } else {
+        // event handlers are no CSS Props!
+        if (name.name.indexOf("on") === 0) return true;
 
-                // First, check for ThemeProvider and update that.
-                const themeProviderIndex = path.node.specifiers.findIndex(
-                    specifier => specifier.local.name === "ThemeProvider"
-                );
+        appendToCss.push({
+          name: t.identifier(name.name),
+          value: value.type === "JSXExpressionContainer" ? value.expression : value,
+        });
+        return false;
+      }
+      return true;
+    });
+    if (appendToCss.length > 0) {
+      if (!cssAttr) {
+        cssAttr = t.jsxAttribute(
+          t.jsxIdentifier("css"),
```

Comment:
Afaik, `<span css={}` is invalid in intrinsic JSX elements. The valid alternative is `<span style={}`.

Reply:
Checked out [this part](https://emotion.sh/docs/css#css-prop) of the emotion docs!

`<span css={styles}/>` will be processed by the emotion babel plugin and turned into `<span className={...}/>`

Reply:
🤔 

Yes, but is it safe to assume that everyone has the babel plugin? I'd rather the codemod give users something that works with base emotion out of the box. 

We could even add a plugin option for users to opt-in to babel-plugin-specific syntax sugar like this. I know @kentcdodds [wanted this](https://github.com/paypal/glamorous/issues/419#issuecomment-397847519) and I agree that this would be a really nice feature to add to this codemod.

Reply:
So what would you suggest for code like:
```jsx
<glamorous.Span css={{":hover": {color: "red"}}}/>
``` 

I guess we could do something like:

```jsx
// opts.withPlugin === true
<span css={{":hover": {color: "red"}}}/>

// opts.withPlugin === false
import {css} from emotion
<span className={css({":hover": {color: "red"}})}/>

```

But for the sake of keeping this PR small, I'd suggest to assume that the plugin is installed for now. We can introduce the `withPlugin` flag in a separate PR

Reply:
Great suggestion.

Philosophically, for keeping the PR small, I'd remove the functionality around CSS props and other plugin-related transformations from _this_ PR altogether and include them all in a more encapsulated PR.

This means that the existing transformations around this functionality ought not be removed, but _moved_ to another PR so this one is a little leaner and simpler, even to review.